### DISCUSSION TITLE
Non-affine displacement implementation

### DISCUSSION THREAD

**Discussion by Leni-Yeo** [idaholab/moose#21286] [Q&A Modules: Phase field] [answered]
Good morning, 
I have a question regarding the type of displacements/deformations that MOOSE applies in the running. I am using either "PresetVelocity' or 'Pressure' boundary condition on top to induce uniaxial-compression with no flux or displacements at the remaining boundaries (disp_y exist at left and right); and both of them are resulting in affine deformation. For example, in case of the attached images from initial to deformed configurations, the two balls should displace downward first to the bottom before deforming. Instead they are simply deforming with the mesh after the first step. How do I apply a cylinder-piston type compression boundary condition to compress the balls with non-affine deformation?
I am already using the Crow app for kernels and such
[Affine_Files.zip](https://github.com/idaholab/moose/files/8893249/Affine_Files.zip)
.

**Comment by GiudGiud** [comment, marked as answer]
@jiangwen84 @laagesen

**Reply by laagesen** [reply]
Afraid I am not much help here, sorry...

**Reply by jiangwen84** [reply]
@SudiptaBiswas should know this better.

**Reply by SudiptaBiswas** [reply]
Are you expecting that the particles would drop to the bottom boundary? I don't think you can do that with the coupled phase-field- mechanics approach. You will have to generate the initial condition you want before hand for that.

**Reply by Leni-Yeo** [reply]
@SudiptaBiswas Hi, yes that's what I initially thought would happen when I set that up, so I thought maybe I was missing something in general. But, I understand what you mean. I will start at an appropriate initial condition as you suggested.
